models/repo.go

```diff
@@ -1275,6 +1275,20 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// Create/Remove git-daemon-export-ok for git-daemon...
+		daemonExportFile := path.Join(repo.RepoPath(), `git-daemon-export-ok`)
+		if repo.IsPrivate {
+			if err = os.Remove(daemonExportFile); err != nil {
```

Comment:
The file is created if it's **public** so the inner blocks should be reversed.
```
if repo.IsPrivate && com.IsExist {
  if os.Remove
} else if !repo.IsPrivate && !com.IsExist {
  if os.Create
}
```
but otherwise I "agree" that we need `com.IsExist` on both... (don't really agree since it's bad practice, but fair enough)